Makefile

```diff
@@ -1,25 +1,183 @@
+
+BUILD := ocamlbuild -j $(PARALLEL_JOBS) -build-dir $(BUILD_DIR) $(BUILD_FLAGS)
+
+MOD_DEPS = $(foreach DEP,$(OCAML_DEPS), --depends $(DEP))
+
+SHELL_DEPS=$(foreach DEP,$(OCAML_PKG_DEPS), ocamlPackages.$(DEP)) \
+           $(foreach DEP,$(OCAML_DEPS), ocamlPackages.$(DEP)) \
+           $(foreach DEP,$(DEPS), $(DEP))
+
+UTOP_MODS = $(foreach DEP,$(OCAML_DEPS),\#require \"$(DEP)\";;)
+UTOP_INIT = $(BUILD_DIR)/init.ml
+
+### Test bits
+TESTS_DIR := $(BUILD_DIR)/tests
+TEST_RUN_SRCS := $(shell find $(SOURCE_DIR) -name "*_tests_run.ml")
+TEST_RUN_EXES := $(notdir $(TEST_RUN_SRCS:%.ml=%))
+TEST_RUN_CMDS := $(addprefix $(TESTS_DIR)/, $(TEST_RUN_EXES))
+TEST_RUN_TARGETS:= $(addprefix run-, $(TEST_RUN_EXES))
+
+# =============================================================================
+# Rules to build the system
+# =============================================================================
+
+.PHONY: all build rebuild metadata install unit-test integ-test test $(TEST_RUN_CMDS)
+
+.PRECIOUS: %/.d
+
+%/.d:
+	mkdir -p $(@D)
+	touch $@
+
+all: build
+
+rebuild: clean all
+
+build:
+	$(BUILD) $(NAME).cma $(NAME).cmx $(NAME).cmxa $(NAME).a $(NAME).cmxs \$(EXTRA_TARGETS)
+
+metadata: build
+	$(TRV) build make-meta --name $(NAME) --target-dir $(LIB_DIR) \
+	--root-file Makefile --semver $(SEMVER) --description-file \
+	'$(DESC_FILE)' $(MOD_DEPS)
+
+# This is only used to help during local opam package
+# development
+opam: build
+	$(TRV) opam make-opam --target-dir $(CURDIR) --name $(NAME) \
+	--semver $(SEMVER) --homepage $(HOMEPAGE) --dev-repo $(DEV_REPO) \
+	--lib-dir $(LIB_DIR) --license $(LICENSE) --author $(AUTHOR) \
+	--maintainer $(AUTHOR) --bug-reports $(BUG_REPORTS) \
+	--build-cmd "make" --install-cmd 'make "install" "PREFIX=%{prefix}%" \
+	"SEMVER=%{vrt:version}%"' --remove-cmd 'make "remove" \
+	"PREFIX=%{prefix}%"' $(BUILD_MOD_DEPS) $(MOD_DEPS)
+
+unpin-repo:
+	opam pin remove -y $(NAME)
+
+pin-repo:
+	opam pin add -y $(NAME) $(CURDIR)
+
+install-local-opam: opam pin-repo
+	opam remove $(NAME); opam install $(NAME)
+
+prepare: build
+	$(TRV) opam prepare --organization $(ORGANIZATION) \
+	--target-dir $(BUILD_DIR) --homepage $(HOMEPAGE) \
+	--dev-repo $(DEV_REPO) --lib-dir $(LIB_DIR) --license $(LICENSE) \
+	--name $(NAME) --author $(AUTHOR) --maintainer $(AUTHOR) \
+	--bug-reports $(BUG_REPORTS) --build-cmd "make" \
+	--install-cmd 'make "install" "PREFIX=%{prefix}%"' \
+	--remove-cmd 'make "remove" "PREFIX=%{prefix}%"' $(BUILD_MOD_DEPS) \
+	$(MOD_DEPS) --description-file '$(DESC_FILE)'
+
+install-extra: build
+	mkdir $(PREFIX)/bin
+	cp $(BUILD_DIR)/lib/vrt.native $(PREFIX)/bin/vrt
+
+install-library: metadata
+	cd $(LIB_DIR); ocamlfind install $(NAME) META `find ./  -name "*.cmi" \
+	-o -name "*.cmo" -o -name "*.o" -o -name "*.cmx" -o -name "*.cmxa" \
+	-o -name "*.cmxs" -o -name "*.a" -o -name "*.cma"`
+
+install: install-library install-extra
+
+submit: prepare
+	opam-publish submit $(BUILD_DIR)/$(NAME).$(SEMVER)
+
+remove:
```

Comment:
This is maybe ambiguous, given `install` runs both `install-library` and `install-extra`, one may reasonably assume that remove would reverse those operations, when it only removes from opam, but does not delete `vrt`. I'd prefer to call this `remove-library` or to have it also `rm -f $(PREFIX)/bin/vrt`,